### PR TITLE
Have Webpacker constant output a deprecation message

### PR DESCRIPTION
**Pull request by Judahmeek**
This change causes a deprecation message to be output anytime a method or constant under `Webpacker` is called.

### CODE REVIEW

**Comment by justin808** [comment]
@Judahmeek Maybe this can be done right before pushing a v8 with other breaking changes?

**Comment by Judahmeek** [comment]
@justin808 If I do it right, then it won't be a breaking change.

**Comment by justin808** [comment]
I understand, but it will be annoying as hell when you have a dependency that is not upgrading. And maybe a breaking performance issue. @Judahmeek

File: lib/shakapacker/webpacker.rb
Comment (ahangarha) [inline lib/shakapacker/webpacker.rb:1]: There is a typo in `to their Shakpacker equivalent`.

**Comment by ahangarha** [comment]
To my understanding, if this PR can bring the same functionality for backward compatibility we already have, it is a much better solution. This requires removing all the existing backward compatibility layers and ensuring everything (typically backward compatibility tests) works as expected.

**Comment by Judahmeek** [comment]
@ahangarha Justin is right that using relying on `method_missing` like this likely has high performance costs: https://www.rubyguides.com/2017/06/costs-of-metaprogramming/

And low performance is generally more annoying to debug then an `uninitialized constant Webpacker` error would be.

**Comment by ahangarha** [comment]
I was not expecting such a drop in performance in the case of `method_missing`!
Interesting!

👍🏾

**Pull request closed without merging**